File: leetcode/1.two-sum.ts
/*
 * @lc app=leetcode id=1 lang=typescript
 *
 * [1] Two Sum
 */

// @lc code=start
function twoSum(nums: number[], target: number): number[] {
  const ht = {};
  const ans = [];

  nums.forEach((v, i) => {
    if (ht[v] && ht[v] > 0) {
      ans.push(i);
      ans.push(ht[v] - 1);
      return;
    }
    ht[target - v] = i + 1;
  });

  ans.sort();
  return ans;
}
// @lc code=end

File: leetcode/1.two-sum.cpp
/*
 * @lc app=leetcode id=1 lang=cpp
 *
 * [1] Two Sum
 */

// @lc code=start
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        int n = nums.size();
        unordered_map<int, int> ht;

        for (int i = 0; i < n; i++) {
            ht[nums[i]] = i;
        }
        for (int i = 0; i < n; i++) {
            int tryFind = target - nums[i];

            if (ht.count(tryFind) > 0 and i != ht[tryFind]) {
                vector<int> ans({ i, ht[tryFind] });
                return ans;
            }
        }

        vector<int> ans;
        return ans;
    }
};
// @lc code=end

